pkg/stanza/fileconsumer/file.go

```diff
@@ -211,8 +211,8 @@ func (m *Manager) makeReader(path string) *Reader {
 		// Empty file, don't read it until we can compare its fingerprint
 		if err = file.Close(); err != nil {
 			m.Errorf("problem closing file %s", file.Name())
-			return nil
 		}
+		return nil
```

Comment:
is there a test that would prevent regressions here?

Reply:
I don't know a reasonable way to test this without a refactoring quite a bit, which is something I'd very much like to do as soon as possible.

Reply:
@codeboten, are you ok with merging this? Hardening this package is an ongoing priority for me but this is a fairly annoying bug and I think the necessary changes to make the fix testable would cause it to persist into the next release.

Reply:
I'd say we should merge this given that it's an obvious bug

Reply:
@djaglowski happy to file a bug to follow up on refactoring and testing this further if that helps, unless there's a ticket for this already.

Reply:
Good idea @atoulme. I've created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23069